Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +351,299 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+        msg = "Could not import the C extension for the angular "\
+              "correlation function for mocks."
+        raise ImportError(msg)
+
+    import numpy as np
+    from Corrfunc.utils import translate_isa_string_to_enum, fix_ra_dec,\
+        return_file_with_rbins, convert_to_native_endian, process_weights
+    from future.utils import bytes_to_native_str
+    import itertools
+    import time
+
+    weights1, weights2 = process_weights(weights1, weights2, RA1, RA2,
+                                         weight_type, autocorr)
+
+    # Ensure all input arrays are native endian
+    RA1, DEC1, weights1, RA2, DEC2, weights2 = [
+        convert_to_native_endian(arr, warn=True) for arr in
+        [RA1, DEC1, weights1, RA2, DEC2, weights2]]
+
+    fix_ra_dec(RA1, DEC1)
+    if autocorr == 0:
+        fix_ra_dec(RA2, DEC2)
+
+    # Passing None parameters breaks the parsing code, so avoid this
+    kwargs = {}
+    for k in ['weights1', 'weights2', 'weight_type', 'RA2', 'DEC2']:
+        v = locals()[k]
+        if v is not None:
+            kwargs[k] = v
+
+    integer_isa = translate_isa_string_to_enum(isa)
+    rbinfile, delete_after_use = return_file_with_rbins(binfile)
+    bin_refs = np.arange(1, maxbinref + 1)
+    if link_in_ra:
+        bin_ref_perms = itertools.product(bin_refs, bin_refs)
+    else:
+        bin_ref_perms = [(1, binref) for binref in bin_refs]
+
+    dtype = np.dtype([(bytes_to_native_str(b'nRA'), np.int),
+                      (bytes_to_native_str(b'nDEC'), np.int),
+                      (bytes_to_native_str(b'avg_time'), np.float),
+                      (bytes_to_native_str(b'sigma_time'), np.float)])
+    all_runtimes = np.zeros(maxbinref ** 3, dtype=dtype)
+    all_runtimes[:] = np.inf
+
+    if autocorr == 0:
+        if RA2 is None or DEC2 is None:
+            msg = "Must pass valid arrays for RA2/DEC2 for "\
+                  "computing cross-correlation."
+            raise ValueError(msg)
+    else:
+        RA2 = np.empty(1)
+        DEC2 = np.empty(1)
+
+    if link_in_ra is True:
+        link_in_dec = True
+
+    if not link_in_dec:
+        msg = "Error: Brute-force calculation without any gridding " \
+              "is forced, as link_in_dec and link_in_ra are both set " \
+              "to False. Please set at least one of link_in_dec, link_in_ra=True " \
+              "to enable gridding along DEC or along both RA and DEC."
+        raise ValueError(msg)
+
+    if not link_in_ra:
+        if verbose:
+            msg = "INFO: Since ``link_in_ra`` is not set, only gridding in declination " \
+                  "Checking with refinements in declination ranging from [1, {}] and a " \
+                  "maximum of {} bins".format(maxbinref, max_cells_per_dim)
+            print(msg)
+
+    for ii, (nRA, nDEC) in enumerate(bin_ref_perms):
+        total_runtime = 0.0
+        total_sqr_runtime = 0.0
+
+        for _ in range(nrepeats):
+            t0 = time.time()
```

Comment:
I prefer [`time.perf_counter()`](https://docs.python.org/3.8/library/time.html#time.perf_counter) for timing code.  Or using [`timeit.default_timer()`](https://docs.python.org/3.8/library/timeit.html#module-timeit), which is the same thing.

Actually, I think using `timeit.timeit()` to time code instead of `time` is generally a good idea, but I think in this case the call would be very verbose with all those arguments to `DDtheta_extn()`!

Reply:
Sure, will adopt `time.perf_counter()` in lieu of `time.time()` then.